### PR TITLE
resolves #124 Possibility to indicate a prefix for Local Storage keys

### LINKED ISSUE #124: Feature : Possibility to indicate a prefix for Local Storage keys
It would be useful to be able to provide a string prefixing the local storage keys. This would make it possible to expose the React Kit from the same domain to several different Tock bots.

### PR DESCRIPTION
**Pull request by Fabilin**
Initial PR description (#125 by @rkuffer):

> To enable dialogs to be saved in local storage when the same domain exposes the React Kit for several different bots, this feature provides the optional parameter localStorageHistory.storagePrefix. The string supplied will be used to prefix the keys used by React Kit to save in local storage (e.g. with localStorageHistory.storagePrefix = "myprefix", userId => myprefix_userId, tockMessageHistory => myprefix_tockMessageHistory, etc.).

I believe said PR shows the limits of configuring everything through `useTock`:
- configuration options must always be passed around as arguments, often through [*prop drilling*](https://react.dev/learn/passing-data-deeply-with-context#the-problem-with-passing-props)
- configuration that is used outside the `Chat` component must be duplicated at every use-site
- global configuration and `Chat` rendering properties get muddled within the latter's props

This PR introduces a new mechanism to configure TOCK: `TockConfig`. It works in much the same way as `TockTheme`, but is attached to `TockContext` rather than `ThemeProvider`. Ultimately, I believe most `Chat` props should migrate to it, but that would be a massive API breaking change.

> [!WARNING]
> This PR fully retires `TockOptions#localStorage` as a boolean alias for `localStorageHistory.enable`. It has been deprecated for over a year so I believe users have had the time to update, but this will mandate a major version bump.

resolves #124
resolves #125

### CODE REVIEW

**Review by kbvi — APPROVED** [review]
I appreciate the way these PR do more generic and configure in a dedicated way.  Maybe a new minor or major version would be interesting each time a bunch of deprecations is removed though.

**Review by rkuffer — COMMENTED** [review]
Not beeing a React developer, i trust your improvements. Thanx for your intervention

File: src/TockContext.tsx [hunk 1]
```diff
@@ -7,15 +7,29 @@ import React, {
   Context,
   useContext,
 } from 'react';
+import deepmerge from 'deepmerge';
+import { PartialDeep } from 'type-fest';
 import { retrieveUserId } from './utils';
 
+export const TockConfigContext: Context<TockConfig | undefined> = createContext<
```
Comment (correi-f) [inline src/TockContext.tsx:14]: I would avoid abbreviation Config -> Configuration

**Review by correi-f — APPROVED** [review]

File: src/TockContext.tsx (hunk 1 above)
Comment (kbvi) [reply src/TockContext.tsx:14]: What about `Setting` ? Less verbose, isn't it?

Comment (Fabilin) [reply src/TockContext.tsx:14]: `TockSettings` sounds good

**Pull request merged**